pkg/query/querier.go

```diff
@@ -55,7 +55,7 @@ type queryable struct {
 
 // Querier returns a new storage querier against the underlying proxy store API.
 func (q *queryable) Querier(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
-	return newQuerier(ctx, q.logger, mint, maxt, q.replicaLabel, q.proxy, q.deduplicate, int64(q.maxSourceResolution/time.Millisecond), q.partialResponse, q.warningReporter), nil
+	return newQuerier(ctx, q.logger, mint, maxt, q.replicaLabel, q.proxy, q.deduplicate, int64(q.maxSourceResolution), q.partialResponse, q.warningReporter), nil
```

Comment:
Quite confusing - hard to tell for me if this makes sense (: 

This has the same unit as mint, maxt int64 right? Maybe making the underlying variable `maxSourceResolutionMiliseonconds` or keeping it time.Duration as long as possible makes sense?

Reply:
Hm.. am I right that is PR was actually wrong? https://github.com/improbable-eng/thanos/pull/1144 And we moved to miliseconds there but we are doing exactly the same here and we missed it? :thinking: 

If yes, let's remove https://github.com/improbable-eng/thanos/blob/2c4f64b1b96907295a7f8e99d8fd64697f0eb12a/pkg/query/api/v1.go#L227 I think it's wrong. All is in time.Duration so moving to miliseconds there does not make sense. It's here that we transition from duration to integer having arbitrary unit (miliseconds) so I think this should stay and last PR should be reverted.

Reply:
And yea, master is broken now in terms of downsampling if I am not wrong?

Reply:
Revert PR:https://github.com/improbable-eng/thanos/pull/1154

Reply:
Perhaps we should move the whole function to return `int64` and in a form appropriate to the caller since as we can see it is hard to make a mental connection between such distant places in the code? Because that's what parsing should do IMHO - we shouldn't do more parsing down the road.

Reply:
> And yea, master is broken now in terms of downsampling if I am not wrong?

I think that's a bit of an overstatement if it has never worked properly 😄

Reply:
Sure but now it does not work even more, so there is regression (:  

Anyway, I just want to avoid confusion, sorry for being quite strict here ): 

Now once we had no regression we can actually make that work and nice. Yes, I am fine with:
* sticking to `maxResolutionMilisec int64` name everywhere.
* parseResolution returning int64 straight away.

Thoughts? (:

Reply:
Indeed that's what I've done. Hopefully it is much more clearer now about what kind of units are being used and that no one in the future will run into this problem like me when I was trying to hunt down one problem but accidentally got caught up in this.